console-framework-client/src/main/java/io/axoniq/console/framework/client/AxoniqConsoleRSocketClient.kt

```diff
@@ -46,46 +60,89 @@ class AxoniqConsoleRSocketClient(
     private val encodingStrategy: RSocketPayloadEncodingStrategy,
     private val executor: ScheduledExecutorService,
     private val nodeName: String,
+    private val heartbeatTimeout: Long = 30
 ) : Lifecycle {
-    private var scheduledReconnector: ScheduledFuture<*>? = null
+    private var maintenanceTask: ScheduledFuture<*>? = null
     private val logger = LoggerFactory.getLogger(this::class.java)
 
-    private lateinit var rsocket: RSocket
-    private var connected = false
+    private var rsocket: RSocket? = null
+    private var lastSentHeartbeat = Instant.EPOCH
+    private var lastReceivedHeartbeat = Instant.now()
+    private var lastConnectionTry = Instant.EPOCH
+    private var connectionRetryCount = 0
+
+    init {
+        registrar.registerHandlerWithoutPayload(Routes.Management.HEARTBEAT) {
+            logger.debug("Received heartbeat from AxonIQ Console. Last one was: {}", lastReceivedHeartbeat)
+            lastReceivedHeartbeat = Instant.now()
+            lastReceivedHeartbeat
+        }
+    }
 
     override fun registerLifecycleHandlers(registry: Lifecycle.LifecycleRegistry) {
         registry.onStart(Phase.EXTERNAL_CONNECTIONS, this::start)
         registry.onShutdown(Phase.EXTERNAL_CONNECTIONS, this::dispose)
     }
 
+    /**
+     * Sends a message to the AxonIQ Console. If there is no connection active, does nothing silently.
+     * The connection will automatically be setup. Losing a few reports is no problem.
```

Comment:
In other words, is it really okay if we drop a message?

Reply:
Yes :) The only vital communication is from the server to the client. Reports can be missed. 
If we would eagerly retry the connection here (instead of the exponential backoff implemented now) we could potentially overwhelm the server